croncat/src/streams/tasks.rs

```diff
@@ -17,9 +20,27 @@ use crate::{
 pub async fn tasks_loop(
     mut block_stream_rx: BlockStreamRx,
     mut shutdown_rx: ShutdownRx,
+    signer: GrpcSigner,
+    block_status: Arc<Mutex<AgentStatus>>,
 ) -> Result<(), Report> {
-    let block_consumer_stream =
-        tokio::task::spawn(async move { while let Ok(_block) = block_stream_rx.recv().await {} });
+    let block_consumer_stream = tokio::task::spawn(async move {
+        while let Ok(_block) = block_stream_rx.recv().await {
+            let locked_status = block_status.lock().await;
+            if *locked_status == AgentStatus::Active {
+                let account_addr = signer.account_id().as_ref();
+                let tasks = signer.get_agent_tasks(account_addr).await.unwrap();
+                if tasks.is_some() {
```

Comment:
I think this was the point we talked about today on the call, where we'll want to modify this to handle multiple tasks. I created an issue for this here: https://github.com/CronCats/croncat-rs/issues/29